lib/elixir/src/elixir_def.erl

```diff
@@ -223,7 +223,7 @@ store_definition(Check, Kind, Meta, Name, Arity, File, Module, Defaults, Clauses
         check_valid_kind(Meta, File, Name, Arity, Kind, StoredKind),
         (Check and StoredCheck) andalso
           check_valid_clause(Meta, File, Name, Arity, Kind, Data, StoredMeta, StoredFile),
-        check_valid_defaults(Meta, File, Name, Arity, Kind, Defaults, StoredDefaults, LastDefaults, LastHasBody),
+        check_valid_defaults(Meta, File, Name, Arity, Kind, Defaults, StoredDefaults, LastDefaults, HasBody, LastHasBody),
```

Comment:
Do you think that is better to break into two functions because of the number of params? With one returning something like `{error, {clauses_with_defaults, {Kind, Name, Arity}}` and the other warning or raising the error?

Reply:
You can try it out as a separate commit and let us know what you prefer. It is your call! Everything else in this commit is perfect.

Reply:
I tried out, but it becomes harder to read. I only wrote all params with underscore on the last function to be easier to reason about it, since `HasBody` was only `_` or `true`.